service/history/ndc/history_importer.go

```diff
@@ -253,19 +254,21 @@ func (r *HistoryImporterImpl) applyNonStartEventsAndSerialize(
 	mutableStateSpec MutableStateInitializationSpec,
 	task replicationTask,
 	createNewBranch bool,
-) ([]byte, error) {
+) ([]byte, bool, error) {
 	prepareBranchFn := r.mutableStateMapper.GetOrCreateHistoryBranch
 	if createNewBranch {
 		prepareBranchFn = r.mutableStateMapper.CreateHistoryBranch
 	}
+
 	mutableState, prepareHistoryBranchOut, err := prepareBranchFn(ctx, wfContext, mutableState, task)
 	if err != nil {
-		return nil, err
+		return nil, false, err
 	} else if !prepareHistoryBranchOut.DoContinue {
-		return r.persistHistoryAndSerializeMutableState(ctx, mutableState, mutableStateSpec)
+		token, err := r.persistHistoryAndSerializeMutableState(ctx, mutableState, mutableStateSpec)
+		return token, false, err
```

Comment:
do you also need to check the error here?

Reply:
That's the exactly place I want to return `false` even if err is nil. Because it will indicate the events are not applied.

Reply:
i see. Does it mean the token will not change in this case? It is a little confusing because it persist the history and return not applied. Consider adding some comment.

Reply:
Sure. Will add some comments. From the API caller perspective, when it got result `token, true, nil`, means the events are applied. When it got result `token, false, nil`, means the events supplied is not applied(because it is deduped), so the caller can decide if it should continue import the following events or not.